pkg/services/ngalert/ngalert.go

```diff
@@ -259,15 +260,36 @@ func (ng *AlertNG) init() error {
 					// Create internal Alertmanager.
 					internalAM, err := factoryFn(ctx, orgID)
 					if err != nil {
-						return nil, err
+						return nil, fmt.Errorf("failed to create internal Alertmanager: %w", err)
 					}
 
 					// Create remote Alertmanager.
 					cfg.OrgID = orgID
 					remoteAM, err := createRemoteAlertmanager(ctx, cfg, ng.KVStore, crypto, autogenFn, m, ng.tracer)
 					if err != nil {
-						moaLogger.Error("Failed to create remote Alertmanager, falling back to using only the internal one", "err", err)
-						return internalAM, nil
+						if remoteSecondaryWithRemoteState {
+							// We can't start the internal Alertmanger without the remote state.
+							return nil, fmt.Errorf("failed to create remote Alertmanager: %w", err)
+						}
+						moaLogger.Warn("Failed to create remote Alertmanager, falling back to using only the internal one", "err", err)
+					}
+
+					if remoteSecondaryWithRemoteState {
+						// Pull and merge the remote Alertmanager state.
+						rs, err := remoteAM.GetRemoteState(ctx)
+						if err != nil {
+							return nil, fmt.Errorf("failed to fetch remote state: %w", err)
+						}
+
+						// The internal Alertmanager should implement the StateMerger interface.
+						sm := internalAM.(notifier.StateMerger)
```

Comment:
please update the Alertmanager interface to include this interface. Runtime casting does not seem to be a good idea.

Reply:
That would require updating all our Alertmanager implementations (remote, mocks) with no-op methods for merging silences and nflog.

I purposefully added these methods to a separate interface to:
1. Not bloat the AM interface with unnecessary methods
2. Only implement the methods in the internal AM implementation
3. Get panics if we somehow try to cast another implementation into a `StateMerger`